src/app/mod.rs

```diff
@@ -1,5 +1,6 @@
 mod state;
 pub use state::*;
 
+pub mod intermediate;
```

Comment:
From what I can tell this state is more of a _startup_ state rather than an _intermediate_ state between two other states, hence I'd call it `startup` or `boot` or something like that. Even if it was an intermediate state, `intermediate` tells you too little about what the state actually does, hence that name isn't very good in my opinion.